### PR TITLE
feat: load bootstrap from apk/disk

### PR DESCRIPTION
**Pull request by rushiiMachine**
Relies on https://github.com/Aliucord/AliucordManager/tree/feat/add-bootstrap and https://github.com/Aliucord/AliucordNative/tree/feat/code-pkg-path-constant

### CODE REVIEW

File: API/hermes/hermes.cpp [hunk 1]
```diff
@@ -1446,11 +1449,84 @@ jsi::Value HermesRuntimeImpl::evaluateJavaScript(
...
-        "bootstrap"
-    );
+
+    auto preBootstrap = std::string("nativeModuleProxy.AliucordNative.getConstants()");
```
Comment (Vendicated) [inline API/hermes/hermes.cpp]: u can
```suggestion
    std::string preBootstrap = "nativeModuleProxy.AliucordNative.getConstants()";
```
```suggestion
    auto preBootstrap = "nativeModuleProxy.AliucordNative.getConstants()"s;
```

File: API/hermes/hermes.cpp [hunk 2]
```diff
@@ -1446,11 +1449,84 @@ jsi::Value HermesRuntimeImpl::evaluateJavaScript(
...
+      int size = ftell(file);
+      fseek(file, 0, SEEK_SET);
+
+      char *data = (char*)malloc(size);
+      if (!fgets(data, size, file)) {
+        ::hermes::hermesLog(
+            "AliuHermes",
+            "Failed to read specified number of bytes: %s",
+            strerror(errno));
+        fclose(file);
```
Comment (Vendicated) [inline API/hermes/hermes.cpp:1486]: should free data here

File: API/hermes/hermes.cpp (hunk 1 above)
Comment (rushiiMachine) [reply API/hermes/hermes.cpp]: gotta love cpp bloated syntax

File: API/hermes/hermes.cpp [hunk 3]
```diff
@@ -1446,11 +1449,84 @@ jsi::Value HermesRuntimeImpl::evaluateJavaScript(
...
+        return jsi::Value::null();
+      }
+      fclose(file);
+
+      bootstrap = std::string(data);
```
Comment (Vendicated) [inline API/hermes/hermes.cpp:1492]:
```suggestion
      bootstrap = std::string(data);
      free(data); // std::string makes a copy
```

File: API/hermes/hermes.cpp [hunk 4]
```diff
@@ -1446,11 +1449,84 @@ jsi::Value HermesRuntimeImpl::evaluateJavaScript(
...
+        ::hermes::hermesLog(
+            "AliuHermes",
+            "Failed to read bootstrap.js from packageCodePath: %s",
+            strerror(errno));
+        return jsi::Value::null();
+      }
+      zip_entry_close(zip);
+      zip_close(zip);
+
+      bootstrap = std::string(::hermes::vm::ASCIIRef((char *)data, size).data());
```
Comment (Vendicated) [inline API/hermes/hermes.cpp]: same as above, you very likely need to free the zip entry, either via free or maybe there's a special zip_free

**Pull request merged**